soh/src/code/z_map_exp.c

```diff
@@ -62,32 +68,13 @@ void Map_SetFloorPalettesData(PlayState* play, s16 floor) {
         interfaceCtx->mapPalette[31] = 1;
     }
 
-    switch (play->sceneNum) {
-        case SCENE_DEKU_TREE:
-        case SCENE_DODONGOS_CAVERN:
-        case SCENE_JABU_JABU:
-        case SCENE_FOREST_TEMPLE:
-        case SCENE_FIRE_TEMPLE:
-        case SCENE_WATER_TEMPLE:
-        case SCENE_SPIRIT_TEMPLE:
-        case SCENE_SHADOW_TEMPLE:
-        case SCENE_BOTTOM_OF_THE_WELL:
-        case SCENE_ICE_CAVERN:
-        case SCENE_DEKU_TREE_BOSS:
-        case SCENE_DODONGOS_CAVERN_BOSS:
-        case SCENE_JABU_JABU_BOSS:
-        case SCENE_FOREST_TEMPLE_BOSS:
-        case SCENE_FIRE_TEMPLE_BOSS:
-        case SCENE_WATER_TEMPLE_BOSS:
-        case SCENE_SPIRIT_TEMPLE_BOSS:
-        case SCENE_SHADOW_TEMPLE_BOSS:
-            for (i = 0; i < gMapData->maxPaletteCount[mapIndex]; i++) {
-                room = gMapData->paletteRoom[mapIndex][floor][i];
-                if ((room != 0xFF) && (gSaveContext.sceneFlags[mapIndex].rooms & gBitFlags[room])) {
-                    Map_SetPaletteData(play, room);
-                }
+    if (SCENEDB_ISDUNGEON(entry)) {
```

Comment:
I want to note something very important for these changes. The switch is on play->sceneNum, but the code inside acts on mapIndex.

The way boss rooms work is that they refer back to the dungeon they are in for data on, mostly, the dungeon map. That is what is stored in mapIndex. When code is acting on data from mapIndex, we retrieve the scene for that index and test it (with SCENEDB_ISDUNGEON). In situations where it acts on data from sceneId, we will test the actual scene and not mapIndex. So, in code like this with both dungeons and boss rooms, if is correct to just have SCENEDB_ISDUNGEON. Other instances with use both SCENEDB_ISDUNGEON and SCENEDB_ISBOSS.